common/src/region.rs

```diff
@@ -406,6 +406,13 @@ impl Default for RegionOptions {
     }
 }
 
+#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+/// Info about extents in a region.
```

Comment:
This data structure gives me bad vibes for a few reasons.

- I have a pre-existing dislike for `Block`, because its `value` field is ambiguous and used in different ways
- It looks like we're usually returning a `Vec<ExtentInfo>`, in which all of the `extent_size.value` must be identical (we don't support volumes with mixed block sizes).  However, this isn't prevented by the data structure!
- Pedantically, this isn't info about an **extent**; it's info about a region / volume / subvolume
- We already have `crucible_client_types::RegionExtentInfo`, which contains the same data as this `struct`

If we want an unambiguous data structure, we could do something like the following:
```rust
pub struct VolumeInfo {
    pub block_size: u64,
    pub volumes: Vec<SubVolumeInfo>,
}

struct SubVolumeInfo {
    pub blocks_per_extent: u64,
    pub extent_count: u32,
}
```

Using this data structure, we're promising that we've checked and all of the block sizes are the same; we would need to add that check to `Volume::query_extent_info`.  Volumes without any subvolumes could still return the top-level `VolumeInfo` with a single value in `volumes` (equivalen to return a single-item `Vec<ExtentInfo>`).